src/pages/cli/graphql/authorization-rules.mdx

```diff
@@ -0,0 +1,368 @@
+
+```graphql
+type Todo
+  @model
+  @auth(rules: [
+    { allow: owner, provider: oidc, identityClaim: "user_id" },
+    { allow: private, provider: oidc },
+    { allow: group, provider: oidc, groupClaim: "user_groups" },
+  ]) {
+  content: String
+}
+```
+
+The example above highlights the supported authorization strategies with `oidc` authorization provider. For `owner` and `group` authorization, you also need to [specify a custom identity and group claim](#configure-custom-identity-and-group-claims).
+
+### Configure custom identity and group claims
+
+`@auth` supports using custom claims if you do not wish to use the default Amazon Cognito-provided "username" or "cognito:groups" claims from your JWT token. This can be helpful if you are using tokens from a 3rd party OIDC system or if you wish to populate a claim with a list of groups from an external system, such as when using a [Pre Token Generation Lambda Trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-pre-token-generation.html) which reads from a database. To use custom claims specify `identityClaim` or `groupClaim` as appropriate like in the example below:
+
+```graphql
+type Post @model
+  @auth(rules: [
+    { allow: owner, identityClaim: "user_id" },
+    { allow: groups, groups: ["Moderator"], groupClaim: "user_groups" }
+  ]) {
+  id: ID!
+  owner: String
+  postname: String
+  content: String
+}
+```
+
+In this example the record owner will check against a `user_id` claim. Similarly, if the `user_groups` claim contains a "Moderator" string then access will be granted.
+
+### How it works
+
+Definition of the `@auth` directive:
+```graphql
+# When applied to a type, augments the application with
+# owner and group-based authorization rules.
+directive @auth(rules: [AuthRule!]!) on OBJECT | FIELD_DEFINITION
+input AuthRule {
+  allow: AuthStrategy!
+  provider: AuthProvider
+  ownerField: String # defaults to "owner" when using owner auth
+  identityClaim: String # defaults to "username" when using owner auth
+  groupClaim: String # defaults to "cognito:groups" when using Group auth
+  groups: [String]  # Required when using Static Group auth
+  groupsField: String # defaults to "groups" when using Dynamic Group auth
+  operations: [ModelOperation] # Required for finer control
+}
+
+enum AuthStrategy { owner groups private public }
+enum AuthProvider { apiKey iam oidc userPools }
+enum ModelOperation { create update delete read }
+```
+
+Authorization rules consists of:
+- **authorization strategy** (`allow`): who the authorization rule applies to
+- **authorization provider** (`provider`): which mechanism is used to apply the authorization rule (API Key, IAM, Amazon Cognito user pool, OIDC)
+- **authorized operations** (`operations`): which operations are allowed for the given strategy and provider. If not specified, `create`, `read`, `update`, and `delete` operations are allowed.
```

Comment:
If the operation list is used all opertions not in the list are denied as a part of deny-by-default.